### PR TITLE
Companion PR for new Maxwell mini-app [maxwell-dev]

### PR DESCRIPTION
**Pull request by mlstowell**

### CODE REVIEW

File: src/electromagnetics.md [hunk 1]
```diff
@@ -286,6 +286,121 @@ Note that this application assumes the mesh coordinates are given in meters.
...
+current (also labeled as $\J$) and an induced current $\sigma\E$.
+
+Solving these equations requires initial conditions for both the
+electric and magnetic fields $\E$ and $\B$ as well as boundary
+conditions related to the tangential components of $\E$ or $\H$.
+Other formulations are possible such as evolving $\H$ and $\D$ or the
+potentials $\varphi$ and $\A$.  This system of equations can also be
+written as a single second order equation involving only $\E$, $\H$,
+$\varphi$, or $\A$.  Each of these formulations has a different set of
+sources, initials and boundary conditions for which it is well-suited.
```
Comment (acfisher) [inline src/electromagnetics.md]: initials->initial

File: src/electromagnetics.md [hunk 2]
```diff
@@ -286,6 +286,121 @@ Note that this application assumes the mesh coordinates are given in meters.
...
+### Maxwell Mini Application
+
+The electrodynamics mini application, named `maxwell` after James Clerk Maxwell who first formulated the classical theory of electromagnetic radiation, is intended to
+demonstrate how to solve transient wave problems in MFEM. Its source
+terms and boundary conditions are simple but they should indicate how more
+specialized sources or boundary conditions could be implemented.
+
+![](img/examples/maxwell-screenshot-small.png)
+
+Time integration is handled by a variable order symplectic time integration
```
Comment (acfisher) [inline src/electromagnetics.md]: Any limitations on the time integration order?  I thought I remember only a few constants for the symplectic time integration.

File: src/electromagnetics.md [hunk 3]
```diff
@@ -286,6 +286,121 @@ Note that this application assumes the mesh coordinates are given in meters.
...
+algorithm.  This algorithm is designed for systems of equations which are
+derived from a Hamiltonian and it helps to ensure energy conservation within
+some tolerance.  The time step used during integration is automatically chosen
+based on the largest stable time step as computed from the largest eigenvalue
+of the update equations.  This determination involves a user-adjustable factor
+which creates a safety margin.  By default the actual time step is less than
+95% of the estimate for the largest stable time step.
+
+Note that this application assumes the mesh coordinates are given in meters.
```
Comment (acfisher) [inline src/electromagnetics.md]: ...and time in seconds?

Comment (mlstowell) [reply src/electromagnetics.md]: Good point.  The code uses seconds but the command line arguments use nanoseconds.  I've added this to the documentation.

File: src/electromagnetics.md (hunk 2 above)
Comment (mlstowell) [reply src/electromagnetics.md]: Yes, the next section describing the command line arguments specifies the allowed values (1-4).

File: src/electromagnetics.md (hunk 1 above)
Comment (mlstowell) [reply src/electromagnetics.md]: good catch

**Comment by tzanio** [comment]
Update also `example.md`

**Comment by tzanio** [comment]
Looks very nice, thank Mark!

The GIF animation is very nice, but also quite large (at least 2MB). Should we consider using a still image instead?

**Comment by mlstowell** [comment]
Yes, it is a bit large.  Let me try to reduce it to quarter size.

**Comment by mlstowell** [comment]
I selected a still image from the original PNG files which I think works better than the JPEG extracted from the GIF.

Would it be ok to leave the original animated GIF if we don't load it by default?  I added a link to the GIF from the electromagnetics.md page.  It is a large file but is it too large if it's optional?

**Comment by tzanio** [comment]
I hardcoded the animation to the mfem.org address so we don't need to include it in the releases when we are exporting the README files. I also compressed the GIF a bit (without losing quality) with https://imageoptim.com/mac.

I think this PR is ready to be merged. Mark, if you agree, do the honors :)

**Pull request merged**